client/blocks/login/login-form.jsx

```diff
@@ -166,16 +196,18 @@ export class LoginForm extends Component {
 						) }
 					</div>
 
-					<div className="login__form-remember-me">
-						<label>
-							<FormCheckbox
-								name="rememberMe"
-								checked={ this.state.rememberMe }
-								onChange={ this.onChangeRememberMe }
-								{ ...isDisabled } />
-							<span>{ this.props.translate( 'Keep me logged in' ) }</span>
-						</label>
-					</div>
+					{ ! this.state.linkingSocialUser && (
+						<div className="login__form-remember-me">
+							<label>
+								<FormCheckbox
+									name="rememberMe"
+									checked={ this.state.rememberMe }
+									onChange={ this.onChangeRememberMe }
+									{ ...isDisabled } />
+								<span>{ this.props.translate( 'Keep me logged in' ) }</span>
+							</label>
+						</div>
+					) }
```

Comment:
I'm not sure there is an actual reason behind it. Maybe they wanted to simplify the form a tiny bit to increase conversion